src/tools/kdb/gen/elektra/structs.cpp

```diff
@@ -0,0 +1,591 @@
+			}
+
+			std::string name = enumValueString;
+			escapeNonAlphaNum (name);
+
+			std::string enumValue = enumPrefix + "_";
+			enumValue += camelCaseToMacroCase (enumValueString);
+			escapeNonAlphaNum (enumValue);
+
+			unionFields.emplace_back (kainjow::mustache::object{ { "name", name },
+									     { "native_type", nativeType },
+									     { "enum_value", enumValue },
+									     { "type_name", typeName },
+									     { "union_type", unionType } });
+		}
+		++i;
+		auto indexString = std::to_string (i);
+		cur = "#" + std::string (indexString.length () - 1, '_') + std::to_string (i);
+	}
+
+	if (unionFields.empty ())
+	{
+		throw CommandAbortException ("Key with alternative references must have at least 1 alternative. Key: " + genKey.getName ());
+	}
+
+	auto name = getName (genKey, fieldKeyName);
+
+	auto field = kainjow::mustache::object{ { "name", name },
+						{ "key_name", keyName },
+						{ "native_type", unionType },
+						{ "type_name", "Union" + unionType },
+						{ "is_array?", isArray },
+						{ "is_struct?", false },
+						{ "is_union?", true },
+						{ "discr_native_type", enumType },
+						{ "discr_type_name", "Enum" + enumType },
+						{ "discr_suffix", discriminatorKeyName },
+						{ "discr_field", discriminatorField } };
+
+	if (isArray)
+	{
+		auto sizeName = arraySizeName (genKey, fieldKeyName);
+
+		fields.emplace_back (kainjow::mustache::object{ { "name", sizeName },
+								{ "is_array_size?", true },
+								{ "array_key", fieldKeyName },
+								{ "native_type", "kdb_long_long_t" } });
+
+		field["size_field"] = sizeName;
+	}
+
+	fields.emplace_back (kainjow::mustache::object{ { "name", discriminatorField },
+							{ "is_discriminator?", true },
+							{ "is_array?", isArray },
+							{ "union_key", fieldKeyName },
+							{ "native_type", enumType } });
+
+	unions.emplace_back (kainjow::mustache::object{ { "native_type", unionType },
+							{ "type_name", "Union" + unionType },
+							{ "fields", unionFields },
+							{ "generate_typedef?", generateUnion },
+							{ "discr_native_type", enumType },
+							{ "default_type", unionFields[0]["native_type"] } });
+
+	return field;
+}
+
+
+kainjow::mustache::object StructFieldsProcessor::processStructRef (const kdb::Key & key, const std::string & keyName,
+								   const std::string & fieldKeyName)
+{
+	std::string typeName;
+	std::string nativeType;
+	std::string restrict;
+	bool alloc = false;
+
+	if (!processStructRef (key, parentKey, allKeys, typeName, nativeType, alloc, restrict))
+	{
+		return processStructRefUnion (key, key, keyName, false, restrict, fieldKeyName);
+	}
+
+	auto name = getName (key, fieldKeyName);
+
+	return kainjow::mustache::object{ { "name", name },	  { "key_name", keyName }, { "native_type", nativeType },
+					  { "type_name", typeName }, { "alloc?", alloc },     { "is_array?", false },
+					  { "is_struct?", true } };
+}
+
+void StructFieldsProcessor::processAll ()
```

Comment:
Some comments for the methods would be nice. processAll does not really tell much.